### PR TITLE
Fixed #35441 -- Fleshed out reference docs for django.template.Context

### PR DESCRIPTION
**Pull request by johnmatthiggins**
# Trac ticket number

ticket-35441

# Branch description
Provide a concise overview of the issue or rationale behind the proposed changes.

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/ref/templates/api.txt [hunk 1]
```diff
@@ -201,10 +201,26 @@ Once you have a compiled :class:`Template` object, you can render a context
 with it. You can reuse the same template to render it several times with
 different contexts.
 
-.. class:: Context(dict_=None)
+.. class:: Context(dict_=None, autoescape=True, use_l10n=None, use_tz=None)
```
Comment (felixxm) [inline docs/ref/templates/api.txt:204]: `RequestContext` signature should also be updated in docs.

File: docs/ref/templates/api.txt [hunk 2]
```diff
@@ -201,10 +201,26 @@ Once you have a compiled :class:`Template` object, you can render a context
...
+      .. warning::
+
+          Only set it to ``False`` if you're rendering non-HTML templates!
+
+    * ``use_l10n`` controls whether blocks will be localized by default. If
+      set to ``True`` numbers and dates will be formatted based on locale.
+    * ``use_tz`` controls whether dates are converted to the
+      local time when rendered in a template. When set to ``True``
+      it forces all dates to be rendered using the local timezone. This takes
+      precedence over :setting:`USE_TZ` and will override it when set to ``True`` or ``False``.
```
Comment (felixxm) [inline docs/ref/templates/api.txt:223]: Wrap docs at 79 chars.

File: docs/ref/templates/api.txt (hunk 1 above)
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt:204]: Updated RequestContext

File: docs/ref/templates/api.txt (hunk 2 above)
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt]: Wrapped text at 79 characters.

**Comment by johnmatthiggins** [comment]
@felixxm Ready for review again!

File: docs/ref/templates/api.txt [hunk 3]
```diff
@@ -201,10 +201,27 @@ Once you have a compiled :class:`Template` object, you can render a context
...
+    * ``autoescape`` controls whether HTML autoescaping is enabled.
+
+      It defaults to ``True``.
+
+      .. warning::
+
+          Only set it to ``False`` if you're rendering non-HTML templates!
+
+    * ``use_l10n`` controls whether blocks will be localized by default. If
+      set to ``True`` numbers and dates will be formatted based on locale.
```
Comment (sarahboyce) [inline docs/ref/templates/api.txt:219]:
```suggestion
      set to ``True`` numbers and dates will be formatted based on locale.

```

File: docs/ref/templates/api.txt [hunk 4]
```diff
@@ -201,10 +201,27 @@ Once you have a compiled :class:`Template` object, you can render a context
...
+    * ``use_tz`` controls whether dates are converted to the
+      local time when rendered in a template. When set to ``True``
+      it forces all dates to be rendered using the local timezone. This takes
+      precedence over :setting:`USE_TZ` and will override it when set to
+      ``True`` or ``False``.
```
Comment (sarahboyce) [inline docs/ref/templates/api.txt:224]:
```suggestion
      precedence over :setting:`USE_TZ`.
```

**Review by sarahboyce — COMMENTED** [review]
Thank you for this @johnmatthiggins I left initial comments 👍

File: docs/ref/templates/api.txt [hunk 5]
```diff
@@ -201,10 +201,27 @@ Once you have a compiled :class:`Template` object, you can render a context
...
+    * ``dict_`` is a dictionary that maps variable names to variable values.
+    * ``autoescape`` controls whether HTML autoescaping is enabled.
+
+      It defaults to ``True``.
+
+      .. warning::
+
+          Only set it to ``False`` if you're rendering non-HTML templates!
+
+    * ``use_l10n`` controls whether blocks will be localized by default. If
```
[the remark this answers is not in the document]
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt]: Applied suggestion.

File: docs/ref/templates/api.txt (hunk 3 above)
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt:220]: Applied suggestion.

File: docs/ref/templates/api.txt [hunk 6]
```diff
@@ -201,10 +201,27 @@ Once you have a compiled :class:`Template` object, you can render a context
...
 
-.. class:: Context(dict_=None)
+.. class:: Context(dict_=None, autoescape=True, use_l10n=None, use_tz=None)
 
-    The constructor of ``django.template.Context`` takes an optional argument —
-    a dictionary mapping variable names to variable values.
+    The constructor of :class:`django.template.Context` takes four optional
+    arguments:
+
+    * ``dict_`` is a dictionary that maps variable names to variable values.
```
[the remark this answers is not in the document]
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt]: Applied fix.

File: docs/ref/templates/api.txt (hunk 4 above)
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt]: Applied suggestion.

[the remark this answers is not in the document]
Comment (johnmatthiggins) [reply docs/ref/templates/api.txt]: That seems like a good idea that will make navigating the documentation easier. I added links to :ref:`topic-l10n-templates` and :ref:`time-zones-in-templates` to the `use_l10n` and `use_tz` parameters. I added references that look like the references found at [`docs/ref/templates/builtins.txt:3001`](https://github.com/django/django/blob/38ad710aba885ad26944ff5708ce1a02a446d2d3/docs/ref/templates/builtins.txt#L3001) and [`docs/ref/templates/builtins.txt:3011`](https://github.com/django/django/blob/38ad710aba885ad26944ff5708ce1a02a446d2d3/docs/ref/templates/builtins.txt#L3011).

**Comment by johnmatthiggins** [comment]
Ready for review again.

**Review by sarahboyce — APPROVED** [review]
Thank you @johnmatthiggins - welcome onboard ⛵

**Pull request merged**